test/endpoint_provisioner_test.go

```diff
@@ -0,0 +1,1105 @@
+					GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(clientResponse.Data).ToNot(BeNil()) // should have a response
+				Expect(clientResponse.Data.QueueName).To(Equal(provisionQueueName))
+				Expect(**(clientResponse.Data.Durable)).To(BeTrue()) // should have provisioned a durable queue on the broker
+
+				// Now let's deprovision the endpoint using the API
+				err = deprovision(provisioner)
+				Expect(err).To(BeNil())
+
+				// check that the endpoint was deprovisioned on the broker via semp
+				clientResponse, _, err = testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+				Expect(err).ToNot(BeNil())
+				Expect(clientResponse.Data).To(BeNil())
+			})
+
+			It("should return error from deprovision when queue does not exist using deprovision function "+deprovisionFunctionName, func() {
+				// check that the endpoint does not yet exist on the broker via semp
+				clientResponse, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+				Expect(err).ToNot(BeNil())
+				Expect(clientResponse.Data).To(BeNil())
+
+				err = deprovision(provisioner) // call to deprovisioner should be unsuccessful
+				Expect(err).ToNot(BeNil())
+				Expect(string(err.Error())).To(Equal("Unknown Queue"))
+
+				// check that we have no endpoint provisioned on the broker via semp
+				clientResponses, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueues(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, nil)
+				Expect(err).To(BeNil())
+				Expect(clientResponses.Data).ToNot(BeNil())
+				Expect(len(clientResponses.Data)).To(Equal(0)) // should have no queues
+			})
+		}
+
+		// invalid queue names to test
+		invalidQueueTestCases := map[string]string{
+			"NameLongerThan201CharactersQueueName": "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ",
+			"ReplayLogQueueName":                   "#REPLAY_LOG_defaultLog",
+			"ReplayTopicsQueueName":                "#REPLAY_TOPICS_defaultLog",
+			"PartitionQueuePrefixQueueName":        "#pq/db583a27835baecd/00011",
+			"TopicLevelsQueueName":                 "/test1/test2/",
+		}
+		for queueLabel, queueName := range invalidQueueTestCases {
+			// TestPlan TestCase Provision#1
+			It("should not provision queue with invalid queueName - "+queueLabel, func() {
+				// remove the provisioned queue
+				defer func() {
+					err := provisioner.Deprovision(queueName, true)
+					Expect(err).ToNot(BeNil())
+					Expect(string(err.Error())).To(Equal("Invalid Queue Name"))
```

Comment:
This is only on cleanup right?
Do you need to assert the error? again? This is just cleanup code right?
Shouldn't the expectation be a nil error, or if this pure cleanup code why expect anything?

Reply:
fixed the unstable tests, removed the unnecessary assertion in the defer function.